### PR TITLE
feat: move shared ports out of experiment

### PR DESCRIPTION
**Pull request by f0ssel**
Completes part of https://github.com/coder/coder/issues/13058

Before with no experiment enabled:
Screenshot 2024-05-02 at 1 51 10 PM

After with enterprise license:
Screenshot 2024-05-02 at 1 54 42 PM

After without license: 
Screenshot 2024-05-02 at 1 57 31 PM

### CODE REVIEW

File: coderd/prometheusmetrics/prometheusmetrics_test.go [hunk 1]
```diff
@@ -511,23 +511,27 @@ func TestAgentStats(t *testing.T) {
 func TestExperimentsMetric(t *testing.T) {
 	t.Parallel()
 
+	if len(codersdk.ExperimentsAll) == 0 {
+		t.Skip("No experiments are currently defined; skipping test.")
+	}
```
Comment (f0ssel) [inline coderd/prometheusmetrics/prometheusmetrics_test.go:516]: @dannykopping I see these prom metrics are somewhat assuming that the `len(codersdk.ExperimentsAll)` will always be at least > 0, but with the removal of shared ports we currently have an empty slice for it and it's failing these tests. Let me know if you'd like me to do something besides skipping these under that condition in this PR. Thanks

Comment (f0ssel) [reply coderd/prometheusmetrics/prometheusmetrics_test.go:516]: We spoke offline, he gave me the 👍 for now and we'll make the changes to get these tests running again soon.

File: coderd/prometheusmetrics/prometheusmetrics_test.go [hunk 2]
```diff
@@ -511,23 +511,35 @@ func TestAgentStats(t *testing.T) {
...
 		experiments codersdk.Experiments
 		expected    map[codersdk.Experiment]float64
 	}{
 		{
 			name:        "Enabled experiment is exported in metrics",
-			experiments: codersdk.Experiments{codersdk.ExperimentSharedPorts},
+			experiments: codersdk.Experiments{
+				// TODO: We need to use a real experiment from codersdk.ExperimentsAll
+				// when the slice is not empty. Uncomment when that happens.
+				// codersdk.ExperimentSharedPorts,
```
Comment (deansheather) [inline coderd/prometheusmetrics/prometheusmetrics_test.go:528]: why not just do `codersdk.ExperimentsAll[0]`?

**Review by deansheather — APPROVED** [review]

File: coderd/prometheusmetrics/prometheusmetrics_test.go (hunk 2 above)
Comment (f0ssel) [reply coderd/prometheusmetrics/prometheusmetrics_test.go:528]: because I'm not as smart as you dean

**Review by stirby — APPROVED** [review]
Transition looks good.

**Pull request merged**